views/static/js/concertScript.js

```diff
@@ -0,0 +1,165 @@
+let concertName = document.currentScript.getAttribute('name');
+let submitButton = document.querySelector('#submit')
+let date = ""
+let gender = "" 
+let concertLocation = ""
+
+console.log(concertName)
+displayAttendButton() 
+getConcert()
+getAttendees(); 
+getLocation()
+
+function displayAttendButton() {
+    console.log("diplay attend")
+    let concertInfo = document.querySelector("#concertInfo")
+    let template = document.querySelector("#attendance")
+    let child = template.content.cloneNode(true)
+    let button = child.querySelector("button")
+
+    // display correct status 
+    let user = "Eunice"
+    let concert = "After Hours Til Dawn Tour"
+
+    fetch(`http://localhost:3000/api/v1/concert/willAttendExists?name=${user}&concert=${concert}`)
+    .then(res => res.json())
+    .then(data => {
+      console.log(data)
+
+      if(data == true) {
+        button.innerText = "Attending"
+      } else {
+        button.innerText = "Attend"
+      }
+      concertInfo.appendChild(child)
+
+      let attendButton = document.querySelector("#attendButton")
+      attendButton.addEventListener("click", (e) => {
+        e.preventDefault();
+        modifyAttend(); 
+    })
+    })
+}
+
+function modifyAttend() { // will this still show up 
+    // let button = document.querySelector("#attendButton")
+    let data = {user: "Eunice", concert: concertName} // update user to current logged in user !!
+    console.log(data)
+
+    if(attendButton.innerText == "Attend") {
+      fetch(`http://localhost:3000/api/v1/concert/willAttend`, {
+        method: 'POST', 
+        headers: {
+        'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(data)
+      })
+      .then( res => res.json())
+      .then(data => {
+        console.log(data)
+        attendButton.innerText = "Attending"
+      })
+    } else if(attendButton.innerText == "Attending") {
+      alert("No longer attending")
+      fetch(`http://localhost:3000/api/v1/concert/deleteAttend`, {
+        method: 'POST', 
+        headers: {
+        'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(data)
+      })
+      .then( res => res.json())
+      .then(data => {
+        console.log(data)
+        attendButton.innerText = "Attend"
+      })
+    }
+  }
+
+  function getConcert() {
+    // query concert data here here 
+    fetch(`http://localhost:3000/api/v1/concert/?name=${concertName}`)
+    .then(res => res.json())
+    .then(data => {
+      console.log(data) 
+
```

Comment:
should we call getLocation() in here? Since this function is called before getLocation() and we won't get concertLocation until then

Reply:
concertLocation sometimes is not showing for me

Reply:
i address this in this PR https://github.com/euniceyoh/CS157C-team7/pull/20